### PR TITLE
SNOW-767766:  Remove _strip_invalid_sf_identifier_chars

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-767766

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Another attempt at addressing the issue described in #758. Test change is taken from original PR.

### CODE REVIEW

File: tests/integ/test_dataframe.py [hunk 1]
```diff
@@ -2085,6 +2085,43 @@ def test_describe(session):
...
 
+    mixed_identifiers_dataframe = session.create_dataframe(
+        data=[
+            [1, Decimal("1.0"), "a", 1, 1, "aa"],
+            [2, Decimal("2.0"), "b", None, 2, "bb"],
+        ],
+        schema=["ほげ", "ふが", "a_ほげ", "ふが_1", "a", "b"],
+    )
+
+    assert mixed_identifiers_dataframe.describe().columns == [
```
Comment (sfc-gh-jdu) [inline tests/integ/test_dataframe.py:2096]: Could we also try several other methods with column names containing special chars? I think it shouldn't only apply to describe.

**Review by sfc-gh-jdu — APPROVED** [review]
We can mention in changelog that we fix this bug

**Review by sfc-gh-sfan — COMMENTED** [review]
I'm curious why we strip these chars before. Now it looks like we just remove all double quotes? 🤔

File: src/snowflake/snowpark/relational_grouped_dataframe.py [hunk 2]
```diff
@@ -49,7 +43,7 @@ def _alias(expr: Expression) -> NamedExpression:
     else:
         return Alias(
             expr,
-            _strip_invalid_sf_identifier_chars(expr.sql.upper()),
+            expr.sql.upper().replace('"', ""),
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/relational_grouped_dataframe.py:46]: @sfc-gh-bli Is this a good fix from your understanding?

Comment (sfc-gh-bli) [reply src/snowflake/snowpark/relational_grouped_dataframe.py:46]: I think, maybe, we should also remove `.replace('"', "")`

Comment (sfc-gh-bli) [reply src/snowflake/snowpark/relational_grouped_dataframe.py:46]: for example, if a table has two columns, one is `"`, the other is `"""`, then we will meet this issue again.

**Comment by sfc-gh-stan** [comment]
Hi all, I've updated the code so the change is protected by client side session parameter `strip_non_ascii_from_agg_cols`.
Please refer to the test cases of `test_df_agg_with_nonascii_column_names` for the result of this change.

**Review by sfc-gh-sfan — APPROVED** [review]

**Review by sfc-gh-yixie — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## 1.4.0 (TBD)
 
+### Behavior Changes
+
+- Aggregate function results (from `DataFrame.agg`, `DataFrame.describe`, etc) no longer strips away non-printing characters from column names.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:7]: Would something like this work?

- Aggregate function results, from functions such as `DataFrame.agg` and `DataFrame.describe`, no longer strip away non-printing characters from column names.

**Pull request merged**